src/main/resources/api/data-repository-openapi.yaml

```
@@ -3904,6 +3904,15 @@ components:
       description: >
         Dataset and snapshot names follow this pattern. It is the same as ObjectNameProperty, but has
         a greater maxLength.
+    ColumnNameProperty:
+      maxLength: 63
+      minLength: 1
+      pattern: ^[a-zA-Z][_a-zA-Z0-9]*$
```

We don't seem to use it, but the regex syntax supports character classes, so this could use the character class for `[_a-zA-Z0-9]` which is `\w`
```suggestion
      pattern: ^[a-zA-Z]\\w*$
```

Also it looks like `ShortIdProperty` below has a bogus regex, it says `[A-za-z0-9_\\-]{22}` but I think it should say `[A-Za-z0-9_\\-]{22}`).

---

Neat!  But I'll keep the existing regex because the resulting error messaging is slightly clearer to the user, especially considering that less tech-savvy users are more likely to run into this problem:

```
{
  "message": "Validation errors - see error details",
  "errorDetail": [
    "schema.tables[0].columns[1].name: 'Pattern' (must match \"^[a-zA-Z]\\w*$\")"
  ]
``` 

Thanks for the note on `ShortIdProperty`.  I think it technically still works, but I'll correct it.

---

I'd say that if a user understands other regex metachars like `^` `$` `*` then expecting them to understand `\w` is reasonable too. Note that the term `\w` has the same meaning in python, javascript and java (and probably others).

Regarding `ShortIdProperty`, the bug was that `[A-z]` includes all the ascii characters between `Z` and `a`, which includes many punctuation chars like `^` and `\` which aren't valid.